torch/nested/_internal/ops.py

```diff
@@ -378,6 +378,70 @@ def is_contiguous_general(func, *args, **kwargs):
 )(is_contiguous_general)
 
 
+@register_jagged_func(
+    torch.ops.aten._to_copy.default,
+    "input: jt_all, dtype: any?, layout: any?, device: any?, pin_memory: any?, non_blocking: any, memory_format: any?",
+)
+def _to_copy_default(func, *args, **kwargs):
+    _, new_kwargs = normalize_function(
+        func, args=args, kwargs=kwargs, normalize_to_only_use_kwargs=True
+    )
+
+    inp: NestedTensor = new_kwargs.pop("input")
+    new_dtype = new_kwargs.pop("dtype", inp.dtype)
+    new_layout = new_kwargs.pop("layout", torch.jagged)
+    new_device: torch.device = new_kwargs.pop("device", inp.device)
+    new_pin_memory = new_kwargs.pop("pin_memory", inp.is_pinned())
+    non_blocking = new_kwargs.pop("non_blocking")
+    new_memory_format = new_kwargs.pop("memory_format", torch.preserve_format)
+    pin_out = non_blocking and inp.is_cuda and new_device.type == "cpu"
+
+    if new_layout not in [torch.strided, torch.jagged]:
+        raise ValueError("Nested Tensors can only have jagged and strided layouts")
+
+    new_values = torch.empty_like(
+        inp.values(),
+        dtype=new_dtype,
+        device=new_device,
+        pin_memory=pin_out,
+        memory_format=new_memory_format,
+    )
+    new_values.copy_(inp.values())
```

Comment:
Why can't you just do something like `inp._values.to_copy(...)`?

Reply:
I was just imitating the code in `NestedTensorFacories.cpp`, but yeah, this should work all the same

Reply:
If you call into `func` no need to mess with the defaults at all